RCOCOA-1883: Add "Sync is not enabled for this App" as a catchable error in error handler (#8643)
realm/realm-swift: CHANGELOG.md

x.y.z Release notes (yyyy-MM-dd)
=============================================================
### Enhancements
* Added support for setting the log level for a category in `Logger`. User will have more fine grained control over 
  the log level for each category.
  ```swift
  Logger.setLogLevel(.info, category: Category.Storage.transactions)
  ```

> * Throw any sync errors tagged with warning action from the server via the sync error handler. 

Reviewer: There's no throwing involved here.

```suggestion
* Report sync warnings from the server such as sync being disabled server-side to the sync error handler. 
```
